Tests/BuiltInRulesTests/PreferKeyPathRuleTests.swift

```diff
@@ -12,6 +17,8 @@ final class PreferKeyPathRuleTests: SwiftLintTestCase {
             .with(nonTriggeringExamples: [
                 Example("f.filter { a in b }"),
                 Example("f.g { $1 }", configuration: Self.extendedMode),
+                Example("f { $0 }", configuration: Self.extendedModeAndIgnoreIdentity),
+                Example("f.map { $0 }", configuration: Self.ignoreIdentity),
```

Comment:
Let's also add a `correction` example to proof that nothing is changed with the option enabled.